guard/tests/parse_tree.rs

```diff
@@ -105,24 +105,43 @@ mod parse_tree_tests {
         S3_BUCKET_LOGGING_ENABLED_PARSE_TREE,
         StatusCode::SUCCESS
     )]
-    #[case("validate/rules-dir/dne.guard", "", StatusCode::INTERNAL_FAILURE)]
+    fn test_yaml_output(
+        #[case] rules_arg: &str,
+        #[case] expected_writer_output: &str,
+        #[case] expected_status_code: i32,
+    ) {
+        let mut writer = Writer::new(WBVec(vec![]));
+        let status_code = ParseTreeTestRunner::default()
+            .rules(rules_arg)
+            .run(&mut writer);
+
+        assert_eq!(expected_status_code, status_code);
+        assert_output_from_str_eq!(expected_writer_output, writer)
+    }
+
+    #[rstest::rstest]
+    #[case(
+        "validate/rules-dir/dne.guard",
+        "Error occurred I/O error when reading No such file or directory (os error 2)\n",
+        StatusCode::INTERNAL_FAILURE
+    )]
     #[case(
         "validate/rules-dir/malformed-rule.guard",
-        "",
+        "Error occurred I/O error when reading No such file or directory (os error 2)\n",
         StatusCode::INTERNAL_FAILURE
     )]
-    fn test_yaml_output(
+    fn test_yaml_output_with_expected_failures(
         #[case] rules_arg: &str,
         #[case] expected_writer_output: &str,
         #[case] expected_status_code: i32,
     ) {
-        let mut writer = Writer::new(WBVec(vec![]));
+        let mut writer = Writer::new_with_custom_stderr(WBVec(vec![]));
```

Comment:
I thought we were going to remove these custom constructors to replace it with a generic one all across the code base, no?